packages/story-editor/src/app/canvas/canvasProvider.js

```diff
@@ -56,6 +58,23 @@ function CanvasProvider({ children }) {
   const [pageCanvasData, setPageCanvasData] = useState(null);
   const [pageCanvasPromise, setPageCanvasPromise] = useState(null);
 
+  const resizeObserver = useMemo(
```

Comment:
Might be useful to add some comments for this as well, looks like it's nicely described in the description of the PR why `useResizeEffect` wouldn't work here, maybe add a line here, too for clarity?

Reply:
Updated this to use an IntersectionObserver so we can get the `clientRect` here which I learned is a different set of coordinates from the `contentRect` you have access too in a resizeObserver. General approach is still the same tho.